pkg/resolution/resolver/bundle/resolver_test.go

```diff
@@ -169,6 +186,322 @@ func TestResolveDisabled(t *testing.T) {
+		}, {
+			name:   "single-task-no-kind",
+			objs:   []runtime.Object{&pipelinev1beta1.Task{TypeMeta: metav1.TypeMeta{APIVersion: "tekton.dev/v1beta1"}, ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			mapper: asIsMapper,
+			args: &args{
+				name: "foo",
+				kind: "task",
+			},
+			expectedStatus:     frtesting.CreateResolutionRequestFailureStatus(),
+			expectedErrMessage: fmt.Sprintf("the layer 0 does not contain a %s annotation", BundleAnnotationKind),
+		}, {
+			name:   "single-task-no-name",
+			objs:   []runtime.Object{&pipelinev1beta1.Task{TypeMeta: metav1.TypeMeta{APIVersion: "tekton.dev/v1beta1", Kind: "task"}}},
+			mapper: asIsMapper,
+			args: &args{
+				name: "foo",
+				kind: "task",
+			},
+			expectedStatus:     frtesting.CreateResolutionRequestFailureStatus(),
+			expectedErrMessage: fmt.Sprintf("the layer 0 does not contain a %s annotation", BundleAnnotationName),
+		}, {
+			name:   "single-task-kind-incorrect-form",
+			objs:   []runtime.Object{&pipelinev1beta1.Task{TypeMeta: metav1.TypeMeta{APIVersion: "tekton.dev/v1beta1", Kind: "Task"}, ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			mapper: asIsMapper,
+			args: &args{
+				name: "foo",
+				kind: "task",
+			},
+			expectedStatus:     frtesting.CreateResolutionRequestFailureStatus(),
+			expectedErrMessage: fmt.Sprintf("the layer 0 the annotation %s must be lowercased and singular, found %s", BundleAnnotationKind, "Task"),
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create a new image with the objects.
+			ref, err := test.CreateImageWithAnnotations(fmt.Sprintf("%s/testbundleresolver/%s", u.Host, tc.name), tc.mapper, tc.objs...)
+			if err != nil {
+				t.Fatalf("could not push image: %#v", err)
+			}
+			// set bundle param to be the image ref
+			tc.args.bundle = ref
+
+			ctx, _ := ttesting.SetupFakeContext(t)
+
+			resolver := &Resolver{}
+			request := createRequest(tc.args.serviceAccount, tc.args.bundle, tc.args.name, tc.args.kind)
+
+			confMap := map[string]string{
+				ConfigKind:           "task",
+				ConfigServiceAccount: "myksa",
+			}
+
+			d := test.Data{
+				ResolutionRequests: []*v1beta1.ResolutionRequest{request},
+				ConfigMaps: []*corev1.ConfigMap{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      ConfigMapName,
+						Namespace: resolverconfig.ResolversNamespace(system.Namespace()),
+					},
+					Data: confMap,
+				}, {
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: resolverconfig.ResolversNamespace(system.Namespace()),
+						Name:      resolverconfig.GetFeatureFlagsConfigName(),
+					},
+					Data: map[string]string{
+						"enable-bundles-resolver": "true",
+					},
+				}},
+			}
+			var expectedStatus *v1beta1.ResolutionRequestStatus
+			var expectedError error
+			if tc.expectedStatus != nil {
+				expectedStatus = tc.expectedStatus.DeepCopy()
+				if tc.expectedErrMessage == "" {
+					reqParams := make(map[string]string)
+					for _, p := range request.Spec.Params {
```

Comment:
Good point. Done. Please take a look.

One possible direction to make it even more clear to see what we expect is to explicitly declare what is expected annotation and source other than the data in the ResolutionRequestStatus in the test table instead of inside the for loop. But I guess the downside is that the test table will be too verbose. Not sure if we want to go down that path. If so, happy to do it in another pr.